frontend/src/bundles/common/components/user-photo-uploader/user-photo-cropper.tsx

```diff
@@ -0,0 +1,122 @@
+import 'cropperjs/dist/cropper.css';
+
+import React, { useCallback,useRef,useState } from 'react';
+import { type ReactCropperElement } from 'react-cropper';
+import { Cropper } from 'react-cropper';
+import { IconContext } from 'react-icons';
+import { BsSave } from 'react-icons/bs';
+import { FaFileUpload } from 'react-icons/fa';
+import { FaArrowRotateRight } from 'react-icons/fa6';
+import { GrZoomIn , GrZoomOut } from 'react-icons/gr';
+
+import styles from './styles.module.scss';
+
+interface UploadCropperProperties {
+    image: string | ArrayBuffer | null;
+    setImage: React.Dispatch<React.SetStateAction<string | ArrayBuffer | null>>
+    setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    setCurrentPhoto: React.Dispatch<React.SetStateAction<string>>;
+    handleImageChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const UserPhotoCropper: React.FC<UploadCropperProperties> = ({ image, setImage, setIsModalOpen, setCurrentPhoto, handleImageChange }) => {
+    const [croppedImage, setCroppedImage] = useState<string | undefined | null>(null);
+    const cropperReference = useRef<ReactCropperElement>(null);
+
+    const handleCrop = useCallback(() => {
+        const croppedCanvas = cropperReference.current?.cropper.getCroppedCanvas();
+        const croppedImageUrl = croppedCanvas?.toDataURL('image/jpeg');
+        setCroppedImage(croppedImageUrl);
+    },[setCroppedImage]);
+
+    // When backend is ready - change handleSave to handleImageChange
+
+    const handleSave = useCallback(() => {
+        if (croppedImage) {
+            setCurrentPhoto(croppedImage);
+        }
+        setIsModalOpen(false);
+    },[croppedImage, setIsModalOpen, setCurrentPhoto]);
+
+    const onNewPhotoClick = useCallback(() => {
```

Comment:
If callback is defined inside component it should have name with prefix handle, change for all callbacks
```suggestion
    const handleNewPhotoClick = useCallback(() => {
```

Reply:
Done